read-fonts/src/tables/truetype/bytecode/decode.rs

```diff
@@ -0,0 +1,237 @@
+//! TrueType bytecode decoder.
+
+use super::{InlineOperands, Instruction, Opcode};
+
+/// An error returned by [`Decoder::decode`] if the end of the bytecode
+/// stream is reached unexpectedly.
+#[derive(Copy, Clone, Debug)]
+pub struct DecodeError(());
+
+impl std::fmt::Display for DecodeError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str("unexpected end of bytecode")
+    }
+}
+
+/// Decodes instructions from TrueType bytecode.
+#[derive(Copy, Clone)]
+pub struct Decoder<'a> {
+    /// The bytecode for the program.
+    pub bytecode: &'a [u8],
+    /// The "program counter" or current offset into the bytecode.
+    pub pc: usize,
```

Comment:
is `pub` necessary here?

Reply:
The control flow instructions will need to read/write `pc`. Calls need to replace `bytecode` if the target function is in a different program. I could probably add methods for this but that just seems like extra complexity for what essentially becomes full access anyway 🤷🏻‍♂️